### PR TITLE
build on linux arm

### PR DESCRIPTION
**Pull request by garyverhaegen-da**

### CODE REVIEW

**Review by carrielaben-da — APPROVED** [review]
Looks good as far as I can tell!

File: daml-lf/language/daml-lf.bzl [hunk 1]
```diff
@@ -110,15 +112,15 @@ LF_DEV_VERSIONS = [
...
+LF_VERSIONS = ([
     "1.8",
     "1.11",
     "1.12",
     "1.13",
     "1.14",
     "1.15",
     "2.1",
-] + LF_DEV_VERSIONS
+] if is_intel else ["1.14", "1.15", "2.1"]) + LF_DEV_VERSIONS
```
Comment (remyhaemmerle-da) [inline daml-lf/language/daml-lf.bzl:123]: What is the motivation for this change ? 
I would be surprise it does not break anything.

**Review by remyhaemmerle-da — APPROVED** [review]
Thanks

File: daml-lf/language/daml-lf.bzl (hunk 1 above)
Comment (remyhaemmerle-da) [reply daml-lf/language/daml-lf.bzl:123]: If the CI is green, we can still merge and investigate that later.

File: bazel_tools/ghc-lib/BUILD.bazel [hunk 2]
```diff
@@ -21,7 +21,7 @@ sh_binaries(
         "@cabal_win//:tools",
         "@dadew_ghc_lib_deps//:tools",
         "@dadew_ghc_lib_deps_python//:tools",
-        "@ghc_865_win//:tools",
+        "@ghc_win//:tools",
```
[the remark this answers is not in the document]
Comment (garyverhaegen-da) [reply bazel_tools/ghc-lib/BUILD.bazel:24]: This specific line is just a rename. In the process of getting things to run on m1, I needed to upgrade the GHC version from 8.6.5 to 9.0.2 on Linux, and decided to do the same on Windows. I could instead rename to `ghc_902_win` if you prefer.

I chose to remove the version from the name to make future upgrades (very slightly) easier.

You can look at https://github.com/digital-asset/daml/pull/18534 to see this one change in isolation.

File: daml-lf/language/daml-lf.bzl (hunk 1 above)
Comment (garyverhaegen-da) [reply daml-lf/language/daml-lf.bzl:123]: The motivation here is pretty simple: this is required for CI to pass on arm. Note that it does indeed break a lot of things, which are also consequently disabled on arm.

The reason this breaks on arm without this change is because the only way we still support 1.8 in the current build is by relying on `damlc_legacy`, which is a 1.18.0-snapshot release that does not have binaries for arm.

Comment (garyverhaegen-da) [reply daml-lf/language/daml-lf.bzl:123]: I went with incremental trial-and-error to decide what gets disabled on arm - it is possible that there is a better way to do this.

Comment (remyhaemmerle-da) [reply daml-lf/language/daml-lf.bzl:123]: make sens.
We probably just have to drop support for old dar. Canton anyway accepts only 1.14 or above. 
I will see that with @moisesackerman-da

Comment (garyverhaegen-da) [reply daml-lf/language/daml-lf.bzl:123]: Thank you!
